### PR TITLE
Vnet integration with App service and Function app

### PR DESCRIPTION
**Pull request by Knyatsine**

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -1,485 +1,565 @@
...
-        "functionsExtensionVersion": {
-            "type": "string",
-            "defaultValue": "~3"
+  "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "resourceEnvironmentName": {
+      "type": "string"
+    },
+    "environmentFullName": {
```
Comment (adam230594) [inline azure/template.json:8]: environmentName

File: azure/template.json [hunk 2]
```diff
@@ -1,485 +1,565 @@
...
+      "type": "string"
+    },
+    "serviceName": {
+      "type": "string"
+    },
+    "sharedResourceGroupName": {
```
Comment (adam230594) [inline azure/template.json:14]: SharedEnvResourceGroup

File: azure/template.json [hunk 3]
```diff
@@ -1,485 +1,565 @@
...
+      "type": "string"
+    },
+    "eventHubPartitionCount": {
+      "type": "int"
+    },
+    "aseHostingEnvironmentName": {
+      "type": "string",
+      "defaultValue": ""
+    },
+    "aseResourceGroup": {
```
Comment (adam230594) [inline azure/template.json:50]: Check what we're passing in here in the pipeline. I don't think this is needed

File: azure/template.json [hunk 4]
```diff
@@ -1,485 +1,565 @@
...
+        },
+        "parameters": {
+          "appServiceName": {
+            "value": "[variables('workerAppServiceName')]"
+          },
+          "appServicePlanName": {
+            "value": "[variables('appServicePlanName')]"
+          },
+          "appServicePlanResourceGroup": {
+            "value": "[resourceGroup().name]"
```
Comment (adam230594) [inline azure/template.json:465]: This will not work

Comment (Knyatsine) [reply azure/template.json:465]: @adam230594  do you mean specifically on this line or the whole block?

File: azure/template.json [hunk 5]
```diff
@@ -1,485 +1,565 @@
...
+    "logAnalyticsWorkspaceName": {
+      "type": "string"
+    },
+    "eventHubNamespaceName": {
+      "type": "string"
+    },
+    "eventHubPartitionCount": {
+      "type": "int"
+    },
+    "aseHostingEnvironmentName": {
```
Comment (adam230594) [inline azure/template.json:46]: Check what we're passing in here in the pipeline. I don't think this is needed

File: azure/template.json [hunk 6]
```diff
@@ -1,17 +1,14 @@
 {
-    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "$schema": "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json",
     "contentVersion": "1.0.0.0",
     "parameters": {
-        "resourceEnvironmentName": {
-            "type": "string"
-        },
-        "environmentFullName": {
+        "environmentName": {
```
Comment (adam230594) [inline azure/template.json:5]: You need resourceEnvironmentName, Its different to environmentName

File: azure/template.json [hunk 7]
```diff
@@ -100,24 +89,82 @@
...
+            "type": "array"
+        },
+        "subnetDelegations": {
+            "type": "array"
         }
     },
     "variables": {
         "deploymentUrlBase": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/",
-        "resourceNamePrefix": "[toLower(concat('das-', parameters('resourceEnvironmentName'),'-', parameters('serviceName')))]",
+        "resourceNamePrefix": "[toLower(concat('das-', parameters('environmentName'),'-', parameters('serviceName')))]",
```
Comment (adam230594) [inline azure/template.json:114]: Should be changed back to resourceEnvironmentName

File: azure/template.json [hunk 8]
```diff
@@ -100,24 +89,82 @@
...
         "databaseName": "[concat(variables('resourceNamePrefix'), '-db')]",
         "appServicePlanName": "[concat(variables('resourceNamePrefix'), '-asp')]",
-        "storageAccountName": "[concat('das', parameters('resourceEnvironmentName'), parameters('serviceName'),'str')]",
+        "storageAccountName": "[concat('das', parameters('environmentName'), parameters('serviceName'),'str')]",
```
Comment (adam230594) [inline azure/template.json:117]: same here resourceEnvironmentName

**Review by adam230594 — APPROVED** [review]

**Pull request merged**